docs/integrations/Webpack.md

```diff
@@ -0,0 +1,195 @@
+# Integration with Webpack
+
+This example shows how you can integrate the MAGE Web SDK with [Webpack](https://webpack.github.io/). Explaining
+Webpack itself is out of scope of this documentation, so please read about Webpack before getting started.
+
+
+## Testing your build configuration
+
+You can make production and development builds using Webpack by running one of the following two commands.
+
+> Production build
+
+```shell
+webpack -p
+```
+
+> Development build
+
+```shell
+webpack -d
+```
+
+If no errors appear, you will know your configuration worked and your project was built into the configured
+build-folder.
+
+
+## Getting started
+
+### Application setup (server)
+
+Below, we will cover two patterns of using Webpack to build your web application using the MAGE Web SDK. In both cases
+We will assume your application was bootstrapped through the MAGE bootstrap process.
+
+That would roughly yield the following file in your project:
+
+> lib/index.js
+
+```javascript
+'use strict';
+
+const mage = module.exports = require('mage');
+
+mage.useModules([
+	'session',
+	'auth',
+	'logger',
+	'time',
+	'archivist'
+]);
+
+mage.useApplicationModules();
+```
+
+### Web SDK setup (browser)
+
+This module configuration is important to the build process, since your build must include configuration for the Web
+SDK. On the browser side, we will expose this configuration as a JSON-string that Webpack can inject using
+`DefinePlugin`. The result is that we can set up the Web SDK as follows:
+
+> www/index.js
+
+```javascript
+window.mageConfig = MAGE_CONFIG;    // Webpack will replace MAGE_CONFIG with a JSON-string
+
+var mage = require('mage-sdk-js');  // mage will automatically pick up window.mageConfig
+```
+
+From here on, we can follow one of two patterns to configure Webpack.
+
+
+## Server and Client code in a separate codebase
+
+### Extracting Web SDK configuration
+
+Your application server will have to share configuration for the build. You can easily write a script that does the
+following:
+
+> scripts/exportClientConfig.js
+
+```javascript
+#!/usr/bin/env node
+
+'use strict';
+
+const fs = require('fs');
+const mage = require('../lib');
+
+const appName = 'game';
+const filePath = '../clientConfig.json';
+
+const mageConfig = mage.getClientConfig(appName);
+
+fs.writeFileSync(filePath, JSON.stringify(mageConfig), { encoding: 'utf8' });
+```
+
+You can then export the configuration to a file:
+
+```shell
+./scripts/exportClientConfig.js
+```
+
+### Configuring Webpack (server)
+
```

Comment:
You might want to add a line to say that you will need to copy the previously generated file to your client app's root directory.